### ISSUE TITLE
HTTP, SOCKS proxy, Tor support

### ISSUE THREAD

**Issue by urlyb** [benbusby/whoogle-search#30]
Hi, 

first of all - great project!

Any chance to add a configurable HTTP, SOCKS proxy support for the requests made towards Google?

Something like this:
```
  if proxy:
            if proxy["type"] == "socks4":
                self.c.setopt(pycurl.PROXYTYPE, pycurl.PROXYTYPE_SOCKS4)
            elif proxy["type"] == "socks5":
                self.c.setopt(pycurl.PROXYTYPE, pycurl.PROXYTYPE_SOCKS5)
            else:
                self.c.setopt(pycurl.PROXYTYPE, pycurl.PROXYTYPE_HTTP)

            self.c.setopt(pycurl.PROXY, str(proxy["address"]))
            self.c.setopt(pycurl.PROXYPORT, proxy["port"])

            if proxy["username"]:
                self.c.setopt(
                    pycurl.PROXYUSERPWD,
                    "{0}:{1}".format(proxy["username"], proxy["password"])
                )

```
This way the search could be even more private.

Thanks!

**Comment by benbusby** [comment]
Thanks! Yeah that's definitely going to be a feature, likely to be implemented at the same time as (or shortly after) #24.

**Comment by benbusby** [comment]
Actually going to merge #24 with this one. It seems like there will be a fair amount of overlap between the two.
